homeassistant/components/egps/config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""ConfigFlow for EGPS devices."""
+import logging
+from typing import Any
+
+from pyegps import get_device, search_for_devices
+from pyegps.exceptions import MissingLibrary, UsbError
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import CONF_DEVICE_API_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFLow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for EGPM devices."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Initiate user flow."""
+
+        if user_input is not None:
+            if CONF_DEVICE_API_ID in user_input:
+                devId = user_input[CONF_DEVICE_API_ID]
+                dev = get_device(device_id=devId)
+                if dev is not None:
+                    await self.async_set_unique_id(dev.device_id)
+                    self._abort_if_unique_id_configured()
+
+                    return self.async_create_entry(
+                        title=f"{devId}",
+                        data={CONF_DEVICE_API_ID: devId},
+                    )
+            return self.async_abort(reason="device_not_found")
+
+        currently_configured = self._async_current_ids(include_ignore=True)
+        try:
+            found_devices = search_for_devices()
```

Comment:
open for discussion: If you are getting all connected devices anyways, wouldn't it be easier for the user to just add all of them in one go and switch the integration's type to hub?

Reply:
It's an interesting idea to let the user add multiple devices at once, but I am not sure how often it will happen that someone adds multiple USB Power-Sockets at the same time. For me a hub is a component which uses one physical connection to manage several devices. In this case each device is connected via USB to the host.